### PR TITLE
Reader Recent Feed Overhaul: Add an empty state for feed and post view

### PR DESCRIPTION
**Pull request by holdercp**
Related to Automattic/loop#193

## Proposed Changes

* Shows an empty state in the post view and styles the post feed
* Show a loading skeleton when new post data is being fetched
* Clean up flickering when switching feeds.

https://github.com/user-attachments/assets/2e7fe161-4429-49a8-a1ef-5249f7d472f0

## Why are these changes being made?

* pe7F0s-2iv-p2

## Testing Instructions

* Make sure you follow a feed with no posts
* Navigate to `/read?flags=reader/recent-feed-overhaul`
* Select the feed with no posts
* Click around different feeds. Loading skeleton should show on first request of the feed but subsequent visits should feel close to instant.
* You might need to throttle your network to make sure a slower connection feels good too.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by eoigal — APPROVED** [review]
LGTM

**Review by DustyReagan — COMMENTED** [review]
It's working, but I'm noticing that the empty state is the default screen while the feed loads. It makes it difficult to tell if I need to continue waiting for the feed to load or if there are actually no posts. Maybe this screen should only appear once we know for sure the feed is empty?


https://github.com/user-attachments/assets/1fdd4cca-a58d-4932-890f-e24c5cd33c56

**Comment by holdercp** [comment]
@DustyReagan Ready for another review. I added a loading state but it presented a lot of flickering as the user clicked around to different feeds. I think I've gotten it all buttoned up and working smoothly.

File: client/reader/recent/index.tsx [hunk 1]
```diff
@@ -166,21 +169,25 @@ const Recent = () => {
 		} ) );
 	}, [ selectedRecentSidebarFeedId ] );
 
+	useLayoutEffect( () => {
```
Comment (holdercp) [inline client/reader/recent/index.tsx:164]: Using a layoutEffect here prevents the flickering when switching feeds. It forces the `isLoading` state to be updated before the browser paints.

File: client/reader/sidebar/reader-sidebar-recent/index.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ const ReaderSidebarRecent = ( {
 	const dispatch = useDispatch();
 
 	let sitesToShow = showAllSites ? sites : sites.slice( 0, SITE_DISPLAY_CUTOFF );
-	const totalUnseenCount = sites.reduce( ( total, site ) => total + site.unseen_count, 0 );
+	// const totalUnseenCount = sites.reduce( ( total, site ) => total + site.unseen_count, 0 );
```
Comment (holdercp) [inline client/reader/sidebar/reader-sidebar-recent/index.tsx:56]: Went ahead and commented out the site unread counts in this PR.

**Review by DustyReagan — APPROVED** [review]
Looks great! 🎉  I still catch a bit of flicker occasionally, but I think we can merge this and follow up if needed.

https://github.com/user-attachments/assets/2b4a8617-b601-47d5-adfc-557d1b69f7c0

**Comment by holdercp** [comment]
I noticed that too but only on my own test feeds, so I thought it has something to do with them being private. That flicker happens because for a few render cycles, there is no data, i.e. `data?.items.length === 0` and `data?.isRequesting === false`. I'm not sure why some feeds end up in that state while others don't.

**Comment by DustyReagan** [comment]
> Since there's still some kinks in the pagination anyway, I'm in favor of removing the pre-fetch to resolve this flash issue for the time being. Thoughts?

Sounds like a good plan to me! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17004671

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @holdercp, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Nothing Posted Yet`
- `This feed is currently empty.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17004671).

**Pull request merged**